### ISSUE TITLE
K8S DNS does't work after simple deployment

### ISSUE THREAD

**Issue by jsuchenia** [openshift/openshift-ansible#4994]
#### Description
After simple installation of Origin DNS inside a kubernetes cluster doesn't work.
I can't ping *kubernetes.default.svc.cluster.local* address neither from a node (any) or from a pod - DNS timeout

##### Version
```
ansible 2.3.1.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jul 18 2017, 09:17:00) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]

$ git describe
openshift-ansible-3.7.1-1-80-g05a9c917

$ git branch
* master

I also tested on branch release-3.6 - the same result
```

##### Steps To Reproduce
1. Install 5 Centos 7 (7.3.1611) machines (sttaic IP, but provided by DHCP)
2. yum install -y NetworkManager
3. systemctl restart NetworkManager.service
2. Deploy 1 master, 1 etcd, 1 router, 2 nodes (5 nodes in total) via command: `ansible-playbook -i inventory playbooks/byo/config.yaml`

##### Expected Results
Describe what you expected to happen.

1. Installation went fine, no errors
2. I can deploy, schedule pods
3. DNS inside kubernetes doesn't work

```
ping kubernetes.default.svc.cluster.local -> works fine (at least resolve)
```

##### Observed Results
Describe what is actually happening.

Ping hangs.... However ping of google.com works fine..

##### Additional Information
System output:
```
[root@os-master ~]# cat /etc/centos-release
CentOS Linux release 7.3.1611 (Core)

[root@os-master ~]# cat /etc/resolv.conf
# Generated by NetworkManager
search lab.internal cloud.lab.internal
nameserver 10.2.0.2
```
Inventory:
```
[OSEv3:vars]
ansible_ssh_user=root
openshift_http_proxy=http://proxy.lab.internal:8080
openshift_https_proxy=http://proxy.lab.internal:8080
openshift_no_proxy=10.0.0.0/8,192.168.0.0/16,172.16.0.0/12,.lab.internal
openshift_deployment_type=origin
openshift_master_identity_providers=[{'name': 'allow_all', 'login': 'true', 'challenge': 'true', 'kind': 'AllowAllPasswordIdentityProvider'}]
openshift_master_cluster_method=native
openshift_master_default_subdomain=10.10.10.2.xip.io
#openshift_hosted_metrics_deploy=true
openshift_disable_check=memory_availability,disk_availability,docker_storage
debug_level=4

[OSEv3:children]
masters
nodes
etcd

[masters]
os-master.cloud.lab.internal

[etcd]
os-master.cloud.lab.internal

[nodes]
os-master.cloud.lab.internal openshift_node_labels="{'region': 'master','zone': 'az0'}" openshift_schedulable=False
os-ingress1-0.cloud.lab.internal openshift_node_labels="{'region': 'infra','zone': 'az1'}"
os-worker1-0.cloud.lab.internal openshift_node_labels="{'region': 'primary','zone': 'az1'}"
os-worker2-0.cloud.lab.internal openshift_node_labels="{'region': 'primary','zone': 'az2'}"
```

The only different configuration is that this lab doesn't have a direct access to the internet (DNS works fine, however only http/https over proxy is allowed)

```
[root@os-master ~]# cat /etc/environment
http_proxy=http://proxy.lab.internal:8080
https_proxy=http://proxy.lab.internal:8080
HTTP_PROXY=http://proxy.lab.internal:8080
HTTPS_PROXY=http://proxy.lab.internal:8080
no_proxy=localhost,127.0.0.0/8,172.16.0.0/12,192.168.0.0./16,10.0.0.0/8,.lab.internal
NO_PROXY=localhost,127.0.0.0/8,172.16.0.0/12,192.168.0.0./16,10.0.0.0/8,.lab.internal
```

**Comment by jsuchenia** [comment]
After installation generated resolv.conf looks like below:

```
[root@os-master ~]# cat /etc/resolv.conf
# nameserver updated by /etc/NetworkManager/dispatcher.d/99-origin-dns.sh
# Generated by NetworkManager
search lab.internal cloud.lab.internal cluster.local
nameserver 10.10.10.1
[root@os-master ~]# ping google.com
PING google.com (216.58.212.78) 56(84) bytes of data.
^C64 bytes from 216.58.212.78: icmp_seq=1 ttl=42 time=5.77 ms

--- google.com ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 5.776/5.776/5.776/0.000 ms
[root@os-master ~]# ping kubernetes.default.svc.cluster.local
ping: kubernetes.default.svc.cluster.local: Name or service not known
```

and dnsmasq configuration:
```
[root@os-master ~]# cat /etc/dnsmasq.d/node-dnsmasq.conf
server=/in-addr.arpa/127.0.0.1
server=/cluster.local/127.0.0.1
[root@os-master ~]# cat /etc/dnsmasq.d/origin-dns.conf
no-resolv
domain-needed
no-negcache
max-cache-ttl=1
enable-dbus
bind-interfaces
listen-address=10.10.10.1
[root@os-master ~]# cat /etc/dnsmasq.d/origin-upstream-dns.conf
server=10.2.0.2
```

**Comment by jsuchenia** [comment]
There is a dnsmasq listening on an external IP:
```
[root@os-master ~]# netstat -nl|grep :53
tcp        0      0 10.2.0.2:53          0.0.0.0:*               LISTEN
udp        0      0 10.2.0.2:53          0.0.0.0:*
```

No other DNS listening on 127.0.0.1

**Comment by sdodson** [comment]
What version of openshift do you have installed?
`openshift version`

I imagine you got 1.5 installed whereas the changes in master and release-3.6 depend on features added in 3.6. If you set openshift_repos_enable_testing=true it will enable the centos testing repo which is the only repo which has 3.6 at the moment.

Alternatively I think just running `yum-config-manager --enable centos-openshift-origin-testing && yum upgrade \*origin\*` and restarting the services will work too.

**Comment by jsuchenia** [comment]
@sdodson Thanks for suggestions, yes - this script is looking for latest RPM version available, which is 1.5.1 currently (and that doesn't work)

Using a setting that you mention enable testing repository and now with 3.6 it's working fine

Thanks for help
